util/reggen/uvm_reg_base.sv.tpl

```diff
@@ -457,29 +457,31 @@ ${_create_reg_field(dv_base_prefix, reg_width, reg_block_path, reg.shadowed, reg
       ${reg_inst}.build(csr_excl);
       default_map.add_reg(.rg(${reg_inst}),
                           .offset(${reg_offset}));
-% if reg.shadowed and reg.hwext:
+% if reg.shadowed:
 <%
-    shadowed_reg_path = ''
-    for tag in reg.tags:
-      parts = tag.split(':')
-      if parts[0] == 'shadowed_reg_path':
-        shadowed_reg_path = parts[1]
+    if reg.hwext:
+      shadowed_reg_path = ''
+      for tag in reg.tags:
+        parts = tag.split(':')
+        if parts[0] == 'shadowed_reg_path':
+          shadowed_reg_path = parts[1]
 
-    if not shadowed_reg_path:
-      print("ERROR: ext shadow_reg does not have tags for shadowed_reg_path!")
-      assert 0
-
-    bit_idx = reg.fields[-1].bits.msb + 1
+      if not shadowed_reg_path:
+        print("ERROR: ext shadow_reg does not have tags for shadowed_reg_path!")
+        assert 0
 
+      bit_idx = reg.fields[-1].bits.msb + 1
 %>\
       ${reg_inst}.add_update_err_alert("${reg.update_err_alert}");
       ${reg_inst}.add_storage_err_alert("${reg.storage_err_alert}");
+  % if reg.hwext:
```

Comment:
Please correct me if I am wrong - 
If we only look for tags when it is external register, then we cannot check `storage error` because we can not probe and force the committed or shadowed value.

Have you tried adding a tag to the shadow register like AES did? https://github.com/lowRISC/opentitan/blob/master/hw/ip/aes/data/aes.hjson#L457

Reply:
but i thought `aes` only needed it because it was an external register and the register is not in its usual location.  Is that right?  For the keymgr, it ends up looking like the following:

Does that work? 
```
      control_shadowed = (keymgr_reg_control_shadowed::
                          type_id::create("control_shadowed"));
      control_shadowed.configure(.blk_parent(this));
      control_shadowed.build(csr_excl);
      default_map.add_reg(.rg(control_shadowed),
                          .offset(32'h14));
      control_shadowed.add_update_err_alert("recov_operation_err");
      control_shadowed.add_storage_err_alert("fatal_fault_err");
      control_shadowed.add_hdl_path_slice(
          "u_reg.u_control_shadowed_start.q",
          0, 1, 0, "BkdrRegPathRtl");
      control_shadowed.add_hdl_path_slice(
          "u_reg.u_control_shadowed_operation.q",
          4, 3, 0, "BkdrRegPathRtl");
      control_shadowed.add_hdl_path_slice(
          "u_reg.u_control_shadowed_cdi_sel.q",
          7, 1, 0, "BkdrRegPathRtl");
      control_shadowed.add_hdl_path_slice(
          "u_reg.u_control_shadowed_dest_sel.q",
          12, 3, 0, "BkdrRegPathRtl");
```

Reply:
wait hold on...something isn't right... let me check again.

Reply:
we actually already have this [here](https://github.com/lowRISC/opentitan/blob/master/util/reggen/uvm_reg_base.sv.tpl#L507), but for whatever reason that's not invoked in my version.  Let me have a closer look.

Reply:
Oh actually you can leave the tags, I will add it later when adding the `shadow_reg_test` automated sequence. CSR tests should work fine without the tags.